core/execution/baremetal/mod.go

```diff
@@ -0,0 +1,37 @@
+package baremetal
+
+import (
+	"go.dedis.ch/dela/core/execution"
+	"go.dedis.ch/dela/core/store"
+	"go.dedis.ch/dela/core/tap"
+	"golang.org/x/xerrors"
+)
+
+// BareMetal is an execution service for packaged applications. Those
+// applications have complete access to the trie and can directly update it.
+//
+// - implements execution.Service
+//
+// TODO: extend to allow registration of other specialized bare-metal services.
+type BareMetal struct {
+	exec execution.Service
+}
+
+// NewExecution returns a new bare-metal execution. The given service will be
+// executed for every incoming transaction.
+func NewExecution(exec execution.Service) BareMetal {
+	return BareMetal{
+		exec: exec,
+	}
+}
+
+// Execute implements execution.Service. It uses the executor to process the
+// incoming transaction and return the result.
+func (bm BareMetal) Execute(tx tap.Transaction, snap store.Snapshot) (execution.Result, error) {
+	res, err := bm.exec.Execute(tx, snap)
```

Comment:
an execution.Service that uses another execution.Service ? Looks like the snake that bites its own tail.

Reply:
Yeah it looks dump right now, but the goal was to register different services (darc, roster, etc...) that are the bare-metal smart contracts and this service takes care of looking up the right one.

Reply:
Which will come while migrating the other packages.